pinot-query-planner/src/test/java/org/apache/pinot/query/queries/ResourceBasedQueryPlansTest.java

```diff
@@ -81,7 +83,7 @@ public void testQueryExplainPlansWithExceptions(String testCaseName, String quer
     }
   }
 
-  @DataProvider
+  @DataProvider(parallel = true)
```

Comment:
what's the reason behind this change? (btw. calcite is [not threadsafe](https://lists.apache.org/thread/wqjjsnyrwxsv9czctkh74v4r8h9w3w2r))

Reply:
Calcite is not problematic here given we create a new instance on each query. It has to be thread safe in that case, otherwise two concurrent queries may affect each other.

This change reduce the time to test in my local computer by half. Given it is a free performance gain, I included it here and I encourage to use parallel whenever is possible.

Reply:
oh. i see. so this runs 2 individual processes? (calcite do share some static objects IIUC)

Reply:
i wasn't sure this is fully verified. if possible let's keep this separate into another PR

Reply:
I think it runs in quite more threads than just two. As said I noticed a large performance gain when running with this option. Neither I or the CI found any issue, but as requested I'll be conservative and remove it

Reply:
Done in e73aac404b600f2944324ae42833c974bf2dfe36